mbed-http.lib

```diff
@@ -0,0 +1 @@
+https://os.mbed.com/teams/sandbox/code/mbed-http/#6daf67a96a91
```

Comment:
Is this lib used (why not sources in the repository) ?

Reply:
I think we have policy that no external dependencies into Mbed OS.

Also, this library is used only once in connectivity tests.. Does not really make sense to include here.

Reply:
Good point. That library is used for a test, and should be removed (that test will not be able to run in CI anyway, and can require additional configuration).